rust-runtime/aws-smithy-fuzz/src/main.rs

```diff
@@ -0,0 +1,947 @@
+
+    for revision in &args.revision {
+        let revision_dir = clone_smithyrs(&workdir, &revision, args.rebuild_local_targets);
+
+        let maven_local_subpath = maven_locals.join(&revision);
+        build_revision(&maven_local_subpath, &revision_dir);
+        let fuzzgen_smithy_build =
+            generate_smithy_build_for_target(&maven_local_subpath, &args, &revision, &revision_dir);
+        let smithy_build_path = workdir.join(format!("smithy-build-{revision}.json"));
+        fs::write(&smithy_build_path, assert_valid_json(&fuzzgen_smithy_build)).unwrap();
+        smithy_build(&workdir, &smithy_build_path);
+    }
+    println!("running smithy build for fuzz harness");
+    smithy_build(&workdir, &fuzzgen_smithy_build_path);
+}
+
+fn generate_smithy_build_for_target(
+    maven_local_subpath: &Path,
+    args: &SetupSmithyArgs,
+    revision: &str,
+    revision_dir: &Path,
+) -> String {
+    let mut context = Context::new();
+    context.insert("maven_local", &maven_local_subpath);
+    context.insert("service", &args.service);
+    context.insert("revision", revision);
+    context.insert("rust_runtime", &revision_dir.join("rust-runtime"));
+    context.insert("dependencies", &args.dependency);
+
+    let fuzzgen_smithy_build = Tera::one_off(
+        include_str!("../templates/smithy-build-targetcrate.jinja2"),
+        &context,
+        false,
+    )
+    .unwrap();
+    assert_valid_json(fuzzgen_smithy_build)
+}
+
+fn generate_smithy_build_json_for_fuzzer(
+    args: &SetupSmithyArgs,
+    fuzz_driver_path: &Path,
+    rust_runtime_for_fuzzer: &Path,
+) -> String {
+    let mut context = Context::new();
+    context.insert("service", &args.service);
+    context.insert("revisions", &args.revision);
+    context.insert("maven_local", &fuzz_driver_path);
+    context.insert("rust_runtime", &rust_runtime_for_fuzzer);
+    context.insert("dependencies", &args.dependency);
+
+    let fuzzgen_smithy_build = Tera::one_off(
+        include_str!("../templates/smithy-build-fuzzer.jinja2"),
+        &context,
+        false,
+    )
+    .unwrap();
+    assert_valid_json(fuzzgen_smithy_build)
+}
+
+/// Run smithy build for a given file
+///
+/// # Arguments
+///
+/// * `workdir`: Path to the working directory (this is where the `build` directory) will be created / used
+/// * `smithy_build_json`: Path to the smithy-build.json file
+fn smithy_build(workdir: impl AsRef<Path>, smithy_build_json: impl AsRef<Path>) {
+    println!(
+        "running smithy build for {}",
+        smithy_build_json.as_ref().display()
+    );
+    // Need to delete classpath.json if it exists to work around small bug in smithy CLI:
+    // https://github.com/smithy-lang/smithy/issues/2376
+    let _ = fs::remove_file(workdir.as_ref().join("build/smithy").join("classpath.json"));
+
+    let home_dir = homedir::my_home().unwrap().unwrap();
+    exec(
+        Command::new("rm")
+            .arg("-r")
+            .arg(format!("{}/.m2", home_dir.display()))
```

Comment:
Perhaps nuke just `~/.m2/repository/software/amazon/smithy/`? Lest we overreach and delete important stuff.